movement/watch_faces/complication/menstrual_cycle_face.c

```diff
@@ -0,0 +1,471 @@
+                        state->dates.bit.first_year = date_period.unit.year;
+                        state->dates.bit.prev_day = date_period.unit.day;
+                        state->dates.bit.prev_month = date_period.unit.month;
+                        state->dates.bit.prev_year = date_period.unit.year;
+                        watch_store_backup_data(state->dates.reg, state->backup_register_dt);
+                        beep(settings);
+                    }
+                    break;
+                case reset:
+                    if (state->reset_tracking) {
+                        reset_tracking(state);
+                        state->reset_tracking = !(state->reset_tracking);
+                        beep(settings);
+                    }
+                    break;
+            }
+            break;
+        case EVENT_ALARM_BUTTON_UP:
+            switch (current_page) {
+                case period_in_num_days:
+                    break;
+                case average_cycle:
+                    break;
+                case peak_fertility_window:
+                    break;
+                case period_is_here:
+                    if (total_days_tracked(state))
+                        state->period_today = !(state->period_today);
+                    break;
+                case first_period:
+                    if (!(state->dates.reg)) 
+                        state->days_prev_period = (state->days_prev_period > 99) ? 0 : state->days_prev_period + 1; // Cycle through pages to quickly reset to 0
+                    break;
+                case reset:
+                    state->reset_tracking = !(state->reset_tracking);
+                    break;
+            }
+            break;
+        case EVENT_TIMEOUT:
+            movement_move_to_face(0);
+            break;
+        default:
+            return movement_default_loop_handler(event, settings);
+    }
+
+    watch_display_string((char *)menstrual_cycle_face_titles[current_page], 0);
+    if (state->dates.reg)
+        watch_set_indicator(WATCH_INDICATOR_SIGNAL); // signal that we are now in a tracking state
+
+    char buf[11];
+    switch (current_page) {
+        case period_in_num_days:
+            sprintf(buf, "%2d", days_till_period(state));
+            if (inside_fert_window(state))
+                watch_set_indicator(WATCH_INDICATOR_BELL);
+            watch_display_string(buf, 4);
+            break;
+        case average_cycle:
+            sprintf(buf, "%2d", state->cycles.bit.average_cycle);
+            watch_display_string(buf, 2);
+            break;
+        case peak_fertility_window:
+            if (event.subsecond % 5 && state->dates.reg) { // blink active for 3 quarter-seconds
+                first_day_fert = get_day_pk_fert(state, first_day);
+                last_day_fert = get_day_pk_fert(state, last_day);
+                sprintf(buf, "Fr%2d To %2d", first_day_fert, last_day_fert); // From: first day | To: last day
```

Comment:
This line gives the following warning:

```
../watch_faces/complication/menstrual_cycle_face.c: In function 'menstrual_cycle_face_loop':
../watch_faces/complication/menstrual_cycle_face.c:431:40: warning: '%2d' directive writing between 2 and 3 bytes into a region of size between 2 and 3 [-Wformat-overflow=]
../watch_faces/complication/menstrual_cycle_face.c:431:30: note: directive argument in the range [0, 255]
../watch_faces/complication/menstrual_cycle_face.c:431:17: note: 'sprintf' output between 11 and 13 bytes into a destination of size 11
```

This isn't actually a problem, since the day number will always be between 1 and 31, but it's good to get rid of the warning anyways — a simple way to do that is to just write:

```c
sprintf(buf, "Fr%2d To %2d", first_day_fert % 100, last_day_fert % 100);
```